### PR TITLE
feat(1.0) rewrite for kong > 1.0

### PR DESCRIPTION
**Pull request by hutchic**
**Todo:**
- [ ] Write a PR to update the docs

**Before merging:**
- [x] Kong 1.1
  - [x] https://github.com/Kong/kong/pull/4378
  - [x] https://github.com/Kong/kong/pull/4286
- [x] https://github.com/Kong/kong-build-tools/pull/29

**After merging:**
- [ ] https://github.com/Kong/kubernetes-sidecar-injector/pull/5
- [ ] Deprecate and remove links to https://github.com/Kong/kong-mesh-dist-kubernetes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,11 +23,36 @@ Kong Enterprise is our powerful offering for larger organizations in need of sec
...
+
+Expose the admin api
+```
+kubectl port-forward -n kong svc/kong-control-plane 8001:80
+curl localhost:8001
+```
+
+Access the proxy
+```
+export HOST=`kubectl get nodes --namespace default -o jsonpath='{.items[0].status.addresses[0].address}'`
```
Comment (james-callahan) [inline README.md]: use `$(......)` instead of backticks

File: kong-control-plane-cassandra.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+          - name: KONG_DATABASE
+            value: cassandra
+          - name: KONG_CASSANDRA_CONTACT_POINTS
+            value: cassandra.default.svc
+          - name: KONG_CASSANDRA_KEYSPACE
+            value: kong
+        command: [ "kong", "migrations", "bootstrap" ]
+      containers:
+      - name: kong-control-plane
+        image: kong:1.1rc1
```
Comment (james-callahan) [inline kong-control-plane-cassandra.yaml]: Do we want to hard code a version here? What about just `kong`?

File: kong-ingress-data-plane-postgres.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  selector:
+    matchLabels:
+      app: kong-ingress-data-plane
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  template:
+    metadata:
```
Comment (james-callahan) [inline kong-ingress-data-plane-postgres.yaml]: Add `k8s.konghq.com/sidecar-inject: false` annotation?

File: kong-control-plane-cassandra.yaml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  labels:
+    app: kong
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kong
+subjects:
+- kind: ServiceAccount
+  namespace: kong
+  name: kong
```
[the remark this answers is not in the document]
Comment (hutchic) [reply kong-control-plane-cassandra.yaml]: I believe this was necessary so it's compatible with https://github.com/Kong/kubernetes-sidecar-injector/ but need @james-callahan to confirm or deny.

A goal of these rewrites is to have a singular place to reference as Kong on k8s yaml without having to `if / else` the docs 🤔

[the remark this answers is not in the document]
Comment (james-callahan) [reply kong-control-plane-cassandra.yaml]: That is correct.

File: kong-control-plane-postgres.yaml [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    app: kong
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kong
+subjects:
+- kind: ServiceAccount
+  namespace: kong
+  name: kong
+---
```
[the remark this answers is not in the document]
Comment (hutchic) [reply kong-control-plane-postgres.yaml]: going to resolve this one let's continue to discuss in the above comment 👍

File: kong-control-plane-cassandra.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (hutchic) [reply kong-control-plane-cassandra.yaml]: @hbagdi is there any harm in including this so we can ensure the kubernetes-sidecar-injector is compatible out of the box even if it's not utilized?

[the remark this answers is not in the document]
Comment (hbagdi) [reply kong-control-plane-cassandra.yaml]: No harm. Adding that to a doc or a comment in YAML would be helpful for future reference.

File: cassandra-statefulset.yaml [hunk 6]
```diff
@@ -91,18 +84,17 @@ spec:
   volumeClaimTemplates:
   - metadata:
       name: cassandra-data
-      annotations:
-        volume.beta.kubernetes.io/storage-class: fast
     spec:
       accessModes: [ "ReadWriteOnce" ]
+      storageClassName: fast
```
Comment (hbagdi) [inline cassandra-statefulset.yaml]: Is this a hard requirement?

I'm not 100% sure how this works but what I've seen is if a k8s cluster can't satisfy this requirement then the statefulset will never even deploy.

**Pull request merged**